src/odemis/odemisd/modelgen.py

```diff
@@ -633,7 +633,7 @@ def get_required_components(self, name):
         ret = set()
 
         attrs = self.ast[name]
-        dependencies = attrs.get("dependencies", {}).values()
+        dependencies = list(attrs.get("dependencies", {}).values())
```

Comment:
Actually, this shouldn't be needed.
However, the code just below is broken (due to the changes to dependencies). I think that instead of appending to `dependencies`, it should add to `ret`.